### PR TITLE
feat(source create): Support --staging-database for Snowflake

### PR DESCRIPTION
**Pull request by spencerwilson**
- Add `--staging-database <NAME>` optional argument when creating a Snowflake source

### CODE REVIEW

**Comment by spencerwilson** [comment]
Current dependencies on/for this PR:
* main
  * **PR #155** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/155" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/patch-tech/dpm/155?utm_source=stack-comment).

File: src/command/source.rs [hunk 1]
```diff
@@ -29,6 +29,12 @@ pub enum CreateSource {
 
         #[arg(long)]
         password: String,
+
+        /// Database which dpm Cloud will use to perform change data capture on
+        /// tables in this source. This value is only used when there exist
+        /// accelerated data packges that access data from this source.
```
Comment (ajmasci) [inline src/command/source.rs:35]: typo: `packges` -> `packages`

File: src/command/source.rs [hunk 2]
```diff
@@ -29,6 +29,12 @@ pub enum CreateSource {
...
+        #[arg(long, value_name = "NAME", default_value = "PATCH")]
```
Comment (ajmasci) [inline src/command/source.rs:36]: Might be useful to add this link in the arg's docstring: https://docs.dpm.sh/sources/snowflake#1-provisioning

File: src/command/source.rs [hunk 3]
```diff
@@ -61,6 +68,7 @@ pub async fn create(cs: &CreateSource) -> Result<()> {
                 database,
                 user,
                 authentication_method: SnowflakeAuthenticationMethod::Password { password },
+                staging_database,
```
Comment (ajmasci) [inline src/command/source.rs:71]: What if this staging DB has not yet been created? Will some code fail in `dpm` Cloud, or `config api`? If so, what feedback does the `dpm` CLI user get?

Comment (spencerwilson) [reply src/command/source.rs:71]: Good questions! Currently the command will succeed and they'll discover their issue later (the initial acceleration of the package version will likely fail). For more info:
- https://linear.app/patch-tech/issue/PAT-4262/during-source-creation-warn-about-non-readiness-for-acceleration
- https://linear.app/patch-tech/issue/PAT-4194/when-source-creation-fails-give-specific-personalized-suggested-fix

**Review by ajmasci — APPROVED** [review]

**Pull request merged**